Декоративные элементы всех страниц

DianaLad8 opened the pull request:
---
:mortar_board: [Декоративные элементы всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/2189703/tasks/16)

:boom: https://htmlacademy-htmlcss.github.io/2189703-sedona-36/8/

DianaLad8 committed:
Меняет правило для фона на краткую запись.

Выровняла звездность отеля позиционированием фона.

DianaLad8 committed:
Оптимизирует иконки

Полная перегруппировка векторных рисунков, другая оптимизация с добавлением атрибута viewBox

DianaLad8 committed:
Добавляет в шапку декоративные элементы

DianaLad8 committed:
Меняет цвет шрифта

Цвет основного шрифта в макете не черный, а #333333

DianaLad8 committed:
Добавляет декоративное подчеркивание в статью-описание Седоны

Выправляет отступы в блоках с изображением.

DianaLad8 committed:
Добавляет декоративные элементы в подвал сайта.

Исправляет векторные иконки для дальнейшей интерактивности.

DianaLad8 committed:
Добавляет интерактивность на главную страницу.

Пока ещё не полностью работают векторные рисунки

DianaLad8 committed:
Добавляет интерактивные кнопки вида в каталоге

DianaLad8 committed:
Добавляет интерактивность карточкам отелей и пагинации

С фокусом и векторными иконками пока непонятно

DianaLad8 committed:
Заменяет бордюры на тень при разных состояниях.

Критерий неподвижности при интерактивности.

DianaLad8 committed:
Добавляет активность фокусу у векторных рисунков в подвале

kamunicorn commented on styles/styles.css:
```diff
@@ -153,64 +201,99 @@ body {
+.navigation-current::before {
+  content: "";
+  position: absolute;
+  width: 75%;
```
![image](https://user-images.githubusercontent.com/38280676/221396029-8253f799-14ba-4b53-baa1-6e9ec02febd1.png)
из-за ширины в процентах не всегда подчеркивание будет соответствовать тексту ссылки

kamunicorn commented on index.html:
```diff
@@ -109,23 +107,23 @@ <h2 class="visually-hidden">Преимущества отдыха в Седон
 
           <ul class="features-list">
-            <li class="features-item">
+            <li class="features-item features-housung">
```
опечатка:
![image](https://user-images.githubusercontent.com/38280676/221396127-79043e3a-f74b-4cff-a0e0-6336a4b5c3d1.png)

kamunicorn commented on styles/styles.css:
```diff
@@ -153,64 +201,99 @@ body {
 
-.selected-quantity {
+.navigation-favor::before {
+  content: "12";
```
Если поменяется количество товаров в избранном, как это поменять и отобразить в верстке?

kamunicorn commented on index.html:
```diff
@@ -26,14 +26,13 @@
-              <a class="navigation-link" href="#">
+            <li class="navigaton-item navigation-favor">
+              <a class="navigation-link " href="#">
                 <span class="visually-hidden">Избранное.</span>
```
Надо подумать, как предоставить для скринридеров информацию, сколько товаров в избранном.

DianaLad8 replied:
Это пока выше моих возможностей. Цифра будет меняться, значит может быть задействован JS?
Пока могу только ```<span class="visually-hidden">Избранное. Вы выбрали 12 гостиниц.</span>```

kamunicorn replied:
Мы в проекте не подключаем javascript. Нам надо предусмотреть, как поменять цифру не через css, а в разметке. Технически, можно поменять цифры и через css, заинлайнив стили: `style="content: '9'"` добавить как атрибут. Но: 1. это дурной тон, 2. с отключенными стилями цифры не будет видно, 3. цифра может быть недоступна скринридерам. Поэтому, лучше сделать в разметке.
Должен быть какой-то спан, в который будем писать цифру.

kamunicorn commented on styles/styles.css:
```diff
@@ -283,17 +385,23 @@ body {
+}
 .features-food {
-  background-color: #ffffff;
+  background: #ffffff url(../images/advantage-icon/food.svg) no-repeat center 81px;
```
👍🏻

kamunicorn commented on styles/styles.css:
```diff
@@ -347,27 +458,35 @@ body {
   padding-left: 20px;
  }
 
+ .email[placeholder] {
```
Почему сделан такой селектор?

DianaLad8 replied:
Переучилась))

kamunicorn commented on styles/styles.css:
```diff
@@ -384,37 +505,68 @@ body {
+.tel:hover,
+.tel:focus {
+  color: #756157;
+  outline: none;
```
оставить браузерную обводку при фокусе

kamunicorn commented on styles/styles.css:
```diff
@@ -153,64 +201,99 @@ body {
+}
 
-.selected-quantity {
+.navigation-favor::before {
```
При наведении на эту штуку нельзя кликнуть по ссылке. А хотелось бы и по этому кружку кликнуть и попасть в избранное.

DianaLad8 replied:
Доделаю

kamunicorn commented on catalog.html:
```diff
@@ -50,7 +49,8 @@
               <li class="breadcrumbs-item">
-                <a class="breadcrumbs-link" href="index.html">
+                <a class="breadcrumbs-link breadcrumbs-link-home" href="index.html">
+                  <svg class="breadcrumbs-home" xmlns="http://www.w3.org/2000/svg" width="13" height="13" aria-hidden="true" focusable="false" fill="none" viewBox="0 0 13 13"><path  d="m6.5.5-6 5.8v6.2h12V6.3L6.5.5Zm5 11h-10V6.7l5-4.8 5 4.8v4.8Z"/><path fill="#fff" d="M4.5 10.5h1V8h2v2.5h1V7h-4v3.5Z"/></svg>
```
Почему решила заинлайнить?

kamunicorn commented on styles/styles.css:
```diff
@@ -445,12 +597,52 @@ body {
+  fill: rgba(255, 255, 255, 0.6);
+}
+.breadcrumbs-home:active {
+  fill: rgba(255, 255, 255, 0.3);
```
не везде поменялся цвет линий
![image](https://user-images.githubusercontent.com/38280676/221398185-faf19096-f1c9-4afe-ab0a-a12fac68adb4.png)

kamunicorn commented on styles/styles.css:
```diff
@@ -536,47 +747,81 @@ border: none;
+  outline: none;
+}
+.select-control:focus {
+  box-shadow: inset 0 0 0 2px #68A2CA;
```
видна одновременно и серая граница, и голубая обводка фокуса
![image](https://user-images.githubusercontent.com/38280676/221398293-df425ead-5083-48df-a3b5-56b299b7e0b4.png)

DianaLad8 replied:
В стайлгайде нет ничего про 
  этот элемент - видимо я перепутала с формами

kamunicorn replied:
про состояния этого элемента:
![image](https://user-images.githubusercontent.com/38280676/221529232-96fb0bd7-9ed3-4581-aaba-08a708721007.png)
только лишнее состояние "текст введен...", у селекта такого нет.
состояния селекта такие же, как и у остальных инпутов. для фокуса должна добавляться браузерная черная обводка, дополнительно к голубой границе

kamunicorn commented on styles/styles.css:
```diff
@@ -501,22 +690,44 @@ border: none;
   margin-bottom: 32px;
   background-color: #82B3D3;
   border: none;
+  outline: 1px solid rgb(230, 24, 117);
```
можно уже убирать обводку, кнопки правильно встали

kamunicorn commented on styles/styles.css:
```diff
@@ -536,47 +747,81 @@ border: none;
 }
 
-
+/* Не отодвигается стрелочка от правого края селекта */
```
Это стандартная стрелка для селекта, как флашки у чекбоксов и точки радиокнопок. Отключить можно через `appearance: none;`. [Про свойство appearance](https://doka.guide/css/appearance/).
А стрелку разместить другим способом.

kamunicorn commented on styles/styles.css:
```diff
@@ -692,31 +945,49 @@ border: none;
+.pagination-link:focus {
+  background-color: #68A2CA;
+  box-shadow: inset 0 0 0 1px #82B3D3;
+  outline: none;
```
Тоже оставить фокус по умолчанию

kamunicorn commented on index.html:
```diff
@@ -141,7 +139,7 @@ <h2 class="subscribe-title heading">Подпишитесь на рассылку
           </div>
             <form class="subscribe-form" action="https://echo.htmlacademy.ru/" method="post">
```
поле ввода и кнопка выглядят не совсем так, как в макете:
![image](https://user-images.githubusercontent.com/38280676/221399087-f2ed93e9-07e1-44c8-9441-105926053bd3.png)
1. не должно быть границы у поля (на странице каталога тоже)
2. примыкающие друг к другу края поля и кнопки должны быть без скругления
3. вместо белого фона у поля ввода должен быть серый цвет

DianaLad8 replied:
Доделаю

kamunicorn commented on styles/styles.css:
```diff
@@ -347,27 +458,35 @@ body {
   background-color: #82B3D3;
   border: none;
+
+  font-stretch: condensed;
```
Не совсем поняла, зачем это. Кажется, тут другой шрифт. Посмотри во вкладке Computed - там должен быть написан шрифт, который применился на элементе.
В макете:
![image](https://user-images.githubusercontent.com/38280676/221399328-4c8d411e-e5fb-4aae-beae-437d9fd21c2c.png)
В верстке:
![image](https://user-images.githubusercontent.com/38280676/221399340-9abe32b7-e111-4c90-ada7-4f0c5040a351.png)

DianaLad8 replied:
Здесь вообще караул. Жду, когда осенит. Пока инсайт до меня не доходит. Потратила 2 дня - бестолку, но хочу сама разобраться.

DianaLad8 replied:
Нашла причину, спасибо за подсказку)) Это браузер во всем виноват :))

kamunicorn commented on styles/styles.css:
```diff
@@ -692,31 +945,49 @@ border: none;
 .pagination-current,
-.dots {
-    background: none;
+.pagination-current:hover,
```
Можно сделать проще: не переназначать стили тут, а при задании стилей состояний для `.pagination-link` исключить текущие через :not

DianaLad8 replied:
Тогда первая кнопка становиться голубой, а если убрать стили состояний, так как нет ссылки, то состояния все равно работают. Я специально повесила класс " текущая", потому что она может меняться на 2 или 10.

kamunicorn commented on styles/styles.css:
```diff
@@ -614,45 +864,42 @@ border: none;
-  background-repeat: repeat-x;
+  background: url(../images/icons/star.svg) repeat-x 0 50%;
 }
 .hotel-four-star {
```
Жду теперь еще классы для звезд 1 и 5

kamunicorn commented on styles/styles.css:
```diff
@@ -445,12 +597,52 @@ body {
+}
+
+.breadcrumbs-link:hover {
+  color: rgba(255, 255, 255, 0.6);
```
Состояния должны применяться только к ссылкам, а Гостиницы - текущая и недоступна для перехода по ссылке.
Понятно, что в макете нарисовано именно для гостиниц, но там показан сам стиль для состояний

DianaLad8 replied:
Исправила

kamunicorn commented on styles/styles.css:
```diff
@@ -445,12 +597,52 @@ body {
 
+.breadcrumbs-link-home {
+  padding-left: 0;
+  padding-right: 18px;
```
Меня смущает, что в фокус попадает и стрелка, хотя она вроде как не относится к ссылке, а разделитель пунктов.
![image](https://user-images.githubusercontent.com/38280676/221400279-a6fe2a34-01b3-4ffd-9611-572b9838f761.png)

DianaLad8 replied:
Здесь получилась грубая ошибка - буду переделывать полностью.

DianaLad8 replied:
Не совсем поняла - заинлайнить. Это про спан? Забыла поменять.

kamunicorn replied:
для домика вставлен svg в разметку, это называется инлайновый свг. его можно убрать в фон ссылки, как и аналогичные элементы в хэдере - избранное, поиск

kamunicorn commented on styles/styles.css:
```diff
@@ -445,12 +597,52 @@ body {
+  content: none;
+}
+
 .breadcrumbs-link {
```
Надо подумать, что случится, если добавится еще одна ссылка:
![image](https://user-images.githubusercontent.com/38280676/221400372-2f1d11da-07fa-4943-be9e-6aeb1ed902dd.png)

kamunicorn replied:
хорошо получилось с хлебными крошками 👍🏻

kamunicorn reviewed (commented):
1. Множественные фоны и просто сокращенное задание фона 🔥
Что переделать:
2. Количество товаров в избранном
3. форма подписки
4. подчеркивание текущей пункта навигации
5. оставляем браузерную обводку для телефона, ссылок пагинации
6. доп. классы для звездности отелей
7. разобраться с хлебными крошками - ссылка домой и дополнительная ссылка

DianaLad8 commented:
Спасибо за подробные комментарии. Все сегодня не успеваю переделать. Что успела, то смогла))

DianaLad8 committed:
Исправляет мелкие декоративные элементы по пунктам

Пункты, перечисленные наставником выполнены - избранное, подписка, "звездность", фокус, подчеркивание текущей страницы. Полностью переделаны хлебные крошки. Выпадающий список оставила на следующее ДЗ.

kamunicorn commented on styles/styles.css:
```diff
@@ -153,64 +206,103 @@ body {
+.navigation-current::before {
+  content: "";
+  position: absolute;
+  width: calc(100% - 32px);
```
🔥

kamunicorn reviewed (commented):
Почти все исправлено 👍🏻, мелочи остались до следующего дз
